compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -563,6 +563,8 @@ checkDefDataType (DefDataType _loc _name _serializable params dataCons) = do
       DataEnum names -> do
         unless (null params) $ throwWithContext EEnumTypeWithParams
         checkUnique EDuplicateConstructor names
+      DataSynonym typ -> do -- TODO(NICK): check for cycles
```

Comment:
Where is the case where we are checking the type of an expression against a type synonym?

Reply:
Not there yet. This PR doesn't extend the haskell type checker.
My idea is that type synonyms will be handled (expanded) by `lookupDataType`.
But there might be a better place. We shall see in the following PR!